### PR TITLE
Add Sidekiq

### PR DESCRIPTION
**Pull request by felixclack**
As per the initial ADR, we have opted to use Sidekiq as the background
worker system.

We would like to send emails via a background process.

### Checklist

- [ ] Attach to Trello card
- [ ] Rebased main
- [ ] Cleaned commit history
- [ ] Tested by running locally

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -24,4 +26,10 @@
   get "/accessibility", to: "static#accessibility"
   get "/cookies", to: "static#cookies"
   get "/privacy", to: "static#privacy"
+
+  devise_scope :staff do
```
Comment (malcolmbaig) [inline config/routes.rb]: Nit: worth co-locating this around line 14 with the other staff / support stuff

File: config/initializers/sidekiq.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+Sidekiq.configure_server do |config|
+  config.redis = { url: ENV.fetch("REDIS_URL", "redis://localhost:6379/1") }
+end
+
+Sidekiq.configure_client do |config|
+  config.redis = { url: ENV.fetch("REDIS_URL", "redis://localhost:6379/1") }
```
Comment (malcolmbaig) [inline config/initializers/sidekiq.rb:8]: Do we need to add REDIS_URL [here](https://github.com/DFE-Digital/access-your-teaching-qualifications/blob/main/Dockerfile#L43) if we're using a fetch in this initializer?

**Review by malcolmbaig — APPROVED** [review]
Couple of non-blocking comments, otherwise LG 👍🏼

File: config/initializers/sidekiq.rb (hunk 2 above)
Comment (felixclack) [reply config/initializers/sidekiq.rb:8]: Not sure about this one. I don't use Docker. Happy to add it but can't verify if it does anything.

Comment (felixclack) [reply config/initializers/sidekiq.rb:8]: Looks like we don't do that in Refer. 🤷

**Pull request merged**